plugins/bcollection/_preferences.py

```diff
@@ -0,0 +1,560 @@
+    """
+    def __init__(self, preferences, widget):
+        """
+            Constructor
+            :param preferences: xlgui.preferences.PreferencesDialog
+            :param widget: Gtk.Widget
+        """
+        super(IconSizePreference, self).__init__(
+            preferences, widget
+        )
+
+        for val in [16, 24, 32, 48]:
+            widget.add_mark(val, Gtk.PositionType.BOTTOM, str(val))
+
+        global _icon_size_preference
+        _icon_size_preference = self
+        _link_icons_widgets()
+
+    # Called by super
+    def _setup_change(self):
+        """
+            Connect events
+            :return: None
+        """
+        widget = self.widget
+        _widgets.connect(
+            widget, ['focus-out-event'], self.change
+        )
+        _widgets.connect(
+            widget, ['change-value'], lambda *_: widget.grab_focus()
+        )
+        _widgets.connect(
+            widget, ['format-value'],
+            lambda _gtk_scale, value: str(int(value)) + ' ' + _('px')
+        )
+
+    # Called by super
+    def _get_value(self):
+        """
+            Get value for preference
+            :return: int
+        """
+        return int(super(IconSizePreference, self)._get_value())
+
+
+class PatternsPreference(
+    _create_base_class(
+        'patterns', widgets.TextViewPreference
+    )
+):
+    """
+        Class that handles 'patterns' preference
+        This is handled as a list of strings
+    """
+    TagClasses = enum(General=0, Name=1, Operator=2, Escape=3)
+
+    def __init__(self, preferences, widget):
+        """
+            Constructor
+            :param preferences:
+            :param widget:
+        """
+        # Define needed data before calling super.__init__
+        self.text_buffer = text_buffer = widget.get_buffer()
+
+        # Define tag classes
+        tag_classes = self.TagClasses
+        self._TAGS = {
+            tag_classes.General:
+                text_buffer.create_tag(None, foreground="Black"),
+            tag_classes.Name:
+                text_buffer.create_tag(None, foreground="Green"),
+            tag_classes.Operator:
+                text_buffer.create_tag(None, foreground="Red"),
+            tag_classes.Escape:
+                text_buffer.create_tag(None, foreground="Black",
+                                       weight=Pango.Weight.BOLD)
+        }
+
+        # super.__init__
+        super(PatternsPreference, self).__init__(
+            preferences, widget
+        )
+
+        self.widgets = _widgets.Widgets(self.builder, ['help_button'],
+                                        help_button=self.__on_clicked)
+
+        # Connect signals
+        text_buffer.connect('changed', self.__on_pattern_textbuffer_changed)
+
+    def __on_clicked(self, *_args):
+        """
+            Opens readme
+            :param _args: ignored
+            :return: None
+        """
+        uri = 'file://' + os.path.join(BASE_DIR, 'README.html')
```

Comment:
This should be something like `'file:' + urllib.pathname2url(...)` to escape special characters correctly and to work on Windows.

Reply:
Nice, thanks! 👍